Add toggle for keymap activation in package settings

tmunro opened the pull request:
Adds UI support for toggling keymap activation on a package-level. See https://github.com/atom/atom/pull/8130 for more details.

<img width="1176" alt="screenshot 2015-07-29 08 58 40" src="https://cloud.githubusercontent.com/assets/3076428/8962553/0e18ccde-35d0-11e5-8214-8df537ea2839.png">

benogle commented:
Can you paste in a screenshot for this?

tmunro commented:
Absolutely, added screenshot.

maxbrunsfeld commented:
Ok, the dependent atom/atom PR is merged, so we can merge this once Atom 1.0.4 has shipped.

This looks good to me UI-wise, though I'd probably tweak the description line (`Whether to use the keybindings from this package`), or possibly just remove that. Maybe @benogle or @thedaniel would have some ideas for copy on this.

@tmunro Could you add a spec for this new button? You could just assert that the package's name is added/removed from the `core.disabledKeymaps` config.

Thanks!

tmunro commented:
Fixed the broken specs and added that new one. One thing I don't quite understand is why calling my new `hasKeymaps` method was breaking the specs. I inlined it for now, but it doesn't really make sense to me.

mnquintana commented on lib/package-keymap-view.coffee:
```diff
@@ -8,21 +9,53 @@ class PackageKeymapView extends View
+      @div class: 'checkbox', =>
+        @label for: 'toggleKeybindings', =>
+          @input id: 'toggleKeybindings', type: 'checkbox', outlet: 'keybindingToggle'
+          @div class: 'setting-title', 'Use Keybindings?'
```
What do you think about dropping the `?` here? I think we tend to avoid punctuation when rendering setting names.

maxbrunsfeld replied:
:+1:

tmunro commented:
Removed the ? from the setting name and changed the description to help explain why you might want to use the setting.

maxbrunsfeld commented:
OK @tmunro, just a heads up that I've merged atom/atom#8173. Sorry for the delay on that one; I was out on vacation.

tmunro commented:
I think this is ready to merge, now that Atom 1.0.8 is out it shouldn't break anything.

thomasjo commented:
Not sure I love the the title of the setting (*Enable*), but someone can always change that later. I'm gonna merge and :ship: this.

Thanks so much for this @tmunro! :heart: :green_heart: :yellow_heart:
